### PR TITLE
Add automatic deletion of offline map regions

### LINKED ISSUE #325: Add automatic deletion of offline map regions
## Description
Add automatic cleanup of offline map regions to prevent storage accumulation. Currently, maps are downloaded when users save/join events but never deleted.

## Problem
- Offline maps persist even after users unsave or leave events
- No cleanup for events that have finished (past their end/start date)
- Storage can fill up with unused map data

## Solution
Implement automatic deletion in three scenarios:
1. When user unsaves an event
2. When user leaves a joined event  
3. When an event finishes (past its end date)

Also increase storage capacity from 2GB to 4GB and max events from 40 to 100 to better support user needs.

## Acceptance Criteria
- [ ] Offline maps deleted when unsaving events
- [ ] Offline maps deleted when leaving joined events  
- [ ] Offline maps deleted for finished events
- [ ] Storage increased to 4GB
- [ ] Max events increased to 100 (saved + joined combined)
- [ ] Unit tests with ≥80% coverage
- [ ] All tests pass
- [ ] Manual testing confirms behavior

### PR DESCRIPTION
**Pull request by ziyad-elarroussi**
## Description
Implements automatic cleanup of offline map regions to prevent storage accumulation and provide better user control over cached data.

**Related Issue:** Closes #325

---

## Changes

**Implementation:**
- Add `observeEventsForDeletion()` to monitor removed and finished events
- Add `removeTileRegion()` method to delete specific regions from TileStore
- Add `isEventFinished()` helper to check if events are past their end/start date
- Store event locations for deletion (eventId → lat/lng mapping)
- Increase TileStore quota from 2GB to 4GB (supports ~100-130 events)
- Increase max regions from 40 to 100 (saved + joined combined)

**Tests:**
- Added 7 new unit tests covering deletion scenarios
- Test deletion when unsaving events
- Test deletion when leaving joined events
- Test deletion for finished events (past endDate)
- Test deletion using start date when no endDate provided
- Test multiple simultaneous deletions
- Updated TileStoreManager tests for 4GB quota

---

## Testing
- [x] Unit tests pass (16 tests total for offline module)
- [ ] Instrumentation tests pass
- [ ] Manual testing completed
- [x] Coverage maintained (≥80%)

**Test summary:** Added 7 unit tests covering all deletion scenarios: unsave, leave, finished events, multiple deletions, and edge cases.

---

## Checklist
- [x] Sufficient documentation and minimal inline comments
- [x] All tests passing
- [x] No new warnings
- [x] If related issue exists: issue has all labels, fields, and milestone filled

### CODE REVIEW

File: app/src/androidTest/java/com/swent/mapin/ui/map/offline/OfflineRegionManagerInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            tileStore = tileStoreManager.getTileStore(), connectivityProvider = { flowOf(true) })
+  }
+
+  @After fun tearDown() = manager.cancelActiveDownload()
+
+  @Test
+  fun removeTileRegion_callsOnCompleteWithSuccess() {
+    val bounds = CoordinateBounds(Point.fromLngLat(6.5, 46.5), Point.fromLngLat(6.6, 46.6))
+    var result: Result<Unit>? = null
+    manager.removeTileRegion(bounds) { result = it }
```
Comment (Acheline) [inline app/src/androidTest/java/com/swent/mapin/ui/map/offline/OfflineRegionManagerInstrumentedTest.kt]: **Nitpick :**
Tests rely on side effects inside callbacks (`result = it`) without asserting callback execution order or ensuring the manager uses the correct dispatcher.
**Suggestion :**
Make the tests coroutine-based (e.g., use `runTest` + a test dispatcher) and force the manager to run on it for deterministic behavior.

File: app/src/main/java/com/swent/mapin/ui/map/offline/EventBasedOfflineRegionManager.kt [hunk 2]
```diff
@@ -168,6 +173,139 @@ class EventBasedOfflineRegionManager(
...
+    // If no dates at all, consider it active
+    return false
+  }
+
+  /**
+   * Deletes tile regions for the given event IDs.
+   *
+   * @param eventIds Set of event IDs to delete regions for
+   */
+  private fun deleteRegionsForEvents(eventIds: Set<String>) {
```
Comment (Acheline) [inline app/src/main/java/com/swent/mapin/ui/map/offline/EventBasedOfflineRegionManager.kt]: **Problem :**
`deleteRegionsForEvents()` launches a coroutine per call, but nested loops inside also launch asynchronous `removeTileRegion` callbacks, making deletion order and completion non-deterministic.
**Suggestion :**
Convert `removeTileRegion` to a suspend function (wrapping callback) and execute deletions sequentially or as supervised concurrent jobs.

File: app/src/test/java/com/swent/mapin/ui/map/offline/EventBasedOfflineRegionManagerTest.kt [hunk 3]
```diff
@@ -299,4 +299,447 @@ class EventBasedOfflineRegionManagerTest {
...
+        }
+
+    every { mockOfflineRegionManager.removeTileRegion(any(), any()) } answers
+        {
+          val onComplete = arg<(Result<Unit>) -> Unit>(1)
+          onComplete(Result.success(Unit))
+        }
+
+    // Create event with endDate in the past (1 hour ago)
+    val pastTime = com.google.firebase.Timestamp(System.currentTimeMillis() / 1000 - 3600, 0)
```
Comment (Acheline) [inline app/src/test/java/com/swent/mapin/ui/map/offline/EventBasedOfflineRegionManagerTest.kt]: **Problem :**
Using `System.currentTimeMillis()` for comparing Firebase timestamps may have timezone drift issues.
**Suggestion :**
Use `Timestamp.now()` or convert all timestamps via the same source to avoid local clock discrepancies.

File: app/src/main/java/com/swent/mapin/ui/map/offline/EventBasedOfflineRegionManager.kt (hunk 2 above)
Comment (ziyad-elarroussi) [reply app/src/main/java/com/swent/mapin/ui/map/offline/EventBasedOfflineRegionManager.kt]: Fixed - converted removeTileRegion to suspend function. Deletions now execute sequentially and deterministically.
